client/src/hooks/useCoverageQuery.ts

```diff
@@ -0,0 +1,40 @@
+import { UndefinedInitialDataOptions, useQuery } from "@tanstack/react-query";
+
+type CoveragePlotData = [number, number, number, number, number][];
+
+interface FetchCoverageResponse {
+  cellType: string;
+  coveragePlot: CoveragePlotData;
+}
+
+async function fetchCoverage(chromosome: string, cellType: string): Promise<FetchCoverageResponse> {
+  const params = new URLSearchParams();
+  params.set("chr", chromosome);
+  params.set("cell_type", cellType);
+
+  const response = await fetch(
+    `/atac/coverage?${params.toString()}`
+  );
+
+  if (!response.ok) {
+    throw new Error(`Error fetching coverage data: ${response.statusText}`);
+  }
+
+  return response.json();
+}
+
+interface UseCoverageQueryOptions {
+  chromosome: string;
+  cellType: string;
+  options?: Partial<UndefinedInitialDataOptions<FetchCoverageResponse, Error>>
+}
+
+export function useCoverageQuery(
+  { chromosome, cellType, options }: UseCoverageQueryOptions
+) {
+  return useQuery<FetchCoverageResponse, Error>({
+    queryKey: ['fetch-coverage', chromosome, cellType],
```

Comment:
Discussion: should we use entity `coverage` as key instead of the actual query name?

This way we'd be able to query by entity and invalidate the cache across all CRUD on `coverage`!

Some notes for context:
https://github.com/chanzuckerberg/single-cell-data-portal/tree/main/frontend/src/common/queries#file-structure

Thank you!